Fix integration tests for JRuby 9.4 and 10.0

tillsc opened the pull request:
This PR fixes the previously broken integration test setup and ensures it now runs successfully with JRuby 9.4 and JRuby 10.

- Integration tests now pass for JRuby 9.4 and 10
- Migrated test app to Rails 7.1
- Updated and cleaned up Ruby-related tooling:
  - Added missing `Gemfile`, `bundle install`, and `rakefile` fixes
  - Pinned `bundler` version in Mavenfile to avoid install drift
- Maven project structure improvements:
  - Switched `<packaging>` to `pom` to avoid generating unnecessary `.jar`/`.war` artifacts
  - Adjusted for manual `target/` directory handling before `warble` build
- Upgraded Jetty:
  - Replaced Jetty 6.1 (Servlet API 2.5) with Jetty 8.1 (Servlet API 3.0)
  - Jetty 8.1 is the last version that allows background startup using `<daemon>true</deamon>`
  - Newer Jetty Maven plugins from Eclipse (versions 9.x–11.x) do not support background execution and are not suitable for this use case

olleolleolle reviewed (approved):
Looks reasonable, I'll start the tests!

headius commented:
Woah nice!

mjansing commented:
I don't know whats happenning on CI.

The test using Java 21 with JRuby 10 (https://github.com/jruby/warbler/actions/runs/16350097899/job/46216486292?pr=567) passes without issues on my local machine.

However, the other integration test fails with Java 8 and JRuby 9.4 on my development machine as well. The error appears to be caused by the ScriptingContainer.class being compiled with Java 21 (class file version 65), which is incompatible with Java 8 (max class file version 52). So the issue likely stems from using a JRuby JAR (jruby-core-*.jar) that was built with a newer Java version than the runtime used during test execution.

olleolleolle commented:
The (21, jruby-9.4, integration) AND the (8, jruby-9.4, integration) fail on: "ERROR:  Error installing /home/runner/.m2/repository/rubygems/bundler/2.7.1/bundler-2.7.1.gem:	bundler-2.7.1 requires Ruby version >= 3.2.0. The current ruby version is 3.1.4."

tillsc commented:
I've now tried adding `Gemfile.lock` files to each existing `Gemfile`. I'm not entirely sure this will work, but since the output shows that `virtus` is being installed — and that gem isn't referenced anywhere in the `integration` subdirectory — it seems like the issue must be caused somewhere outside the integration tests.

tillsc commented:
That didn’t work either. The only thing that seems to have any effect is adding `gem 'bundler', '${bundler.version}'` to the Mavenfile. Without it, some Maven plugin apparently tries to install the latest Bundler version during the JRuby 9.4 integration tests — and fails immediately. For some reason, I haven’t been able to reproduce this behavior locally.

However, adding `gem 'bundler', '${bundler.version}'` breaks one of the regular spec tests for JRuby 10 that previously passed — and I have no idea why.

Also, as @mjansing already pointed out, the JRuby 9.4 integration tests still fail when run with Java 8.

So unfortunately, I can't get all tests to pass at the same time — sorry!

headius commented:
I'm out of the office quite a bit this week but once I'm back at work I'll see if I can help iron out these remaining issues.

tillsc commented:
Sorry for the repeated force pushes — I just didn’t want to mess up the Git history.
I was on the wrong track initially: committing the Gemfile.lock files fixed some surface issues but not the actual problem.
So I started over and explicitly pinned jruby-jars and bundler versions in both the Mavenfile and integration/pom.xml.

As of now, the integration tests are always run with Bundler 2.6.3 and JRuby 9.4, even if JRuby 10 is used externally.
That’s intentional for the time being and can easily be changed later — the versions are controlled via Maven properties, so it’s just a matter of configuration.

Since the integration tests don’t have any Gemfile.lock files, this setup should be fine for now.

tillsc commented:
I added logic to ensure that the currently active JRuby and Bundler versions are also used for the dependencies in the integration tests. That’s good enough for now from my side.

@headius had removed support for running tests with different Bundler versions in the GitHub Actions configuration — I’ve restored that and added an example entry using Bundler 2.7.1 with JRuby 10.

Since we’ve experienced issues in some of our projects when using Bundler versions other than 2.6.3, this setup might help us reproduce those problems during testing. For now, though, everything is green.

kares reviewed (approved):
:heart_decoration: awesome work!
